src/commonMain/kotlin/com/charleskorn/kaml/Annotations.kt

```diff
@@ -32,3 +32,38 @@ import kotlinx.serialization.SerialInfo
 public annotation class YamlComment(
     vararg val lines: String,
 )
+
+/**
+ * Write a String value if it is a single line in the specified ScalarStyle.
+ * This overrides the value specified in the [YamlConfiguration].
+ */
+@OptIn(ExperimentalSerializationApi::class)
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.BINARY)
+@SerialInfo
+public annotation class YamlWriteSingleLineStringUsingScalarStyle(
+    val stringScalarStyle: StringScalarStyle,
+)
+
+/**
+ * Write a String value if it is a multiline in the specified ScalarStyle.
+ * This overrides the value specified in the [YamlConfiguration].
+ */
+@OptIn(ExperimentalSerializationApi::class)
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.BINARY)
+@SerialInfo
+public annotation class YamlWriteMultiLineStringUsingScalarStyle(
+    val stringScalarStyle: StringScalarStyle,
```

Comment:
Any reason not to use the same enum `YamlConfiguration` uses?

Reply:
Yes, that seems like a good option. This evolved from a different initial proposal.

I would propose then to add the `Folded` to the `MultiLineStringStyle` to make it complete.

Reply:
Should I add the Folded here in this MR or make it a separate thing?

Reply:
Adding it as part of this PR is fine.

Reply:
FYI: I do see that the `Literal` and `Folded` formats are not possible for SingleLine cases (which is fine for me and is the same as the current situation).

Reply:
Using existing enums, added Folded